docs/sphinx/dev/dev_dot_product_example.rst

```diff
@@ -0,0 +1,74 @@
+********************
+Dot Product Example
```

Comment:
I'm tempted to rename this just `Example Pipeline`.  The fact it's a dot product shouldn't really be important to the reader, who very likely wants to just learn how to do something like it.